packages/provider/src/index.ts

```diff
@@ -0,0 +1,193 @@
+import type { StacksProvider } from '@stacks/connect';
+
+import { type LeatherRpcMethodMap, type RpcParameter, RpcRequests } from '@leather.io/rpc';
+
+import { addLeatherToProviders } from './add-leather-to-providers';
+import type { WalletMethodNames, WalletResponses } from './rpc-methods';
+
+// TODO: change this to normal env variables
+const BRANCH = 'TEST_BRANCH';
+const COMMIT_SHA = 'TEST_COMMIT_SHA';
+const VERSION = 'VERSION';
+
+function isRN(): boolean {
+  return !!(window as any)?.ReactNativeWebView;
+}
+
+function dispatchCallback(rpcRequest: RpcRequests) {
+  // if running in mobile app
+  if (isRN()) {
+    (window as any).ReactNativeWebView.postMessage(JSON.stringify(rpcRequest)); // RN
+    return;
+  }
+  // if running in extension
+  document.dispatchEvent(new CustomEvent('request', { detail: rpcRequest }));
+}
```

Comment:
This isn't quite what I meant. Here, we're handling platform-specific behaviours _inside_ `@leather.io/providers`.

Why not pass in the platform-specific stuff as callbacks? e.g.

```ts
// extension repo
import { initInpageProvider } from '@leather.io/providers';

initInpageProvider({ 
  onDispatch(event) {
    document.dispatchEvent(new CustomEvent('request', { detail: rpcRequest }));
  }
})

// mobile repo
import { initInpageProvider } from '@leather.io/providers';
initInpageProvider({ 
  onDispatch(event) {
    (window as any).ReactNativeWebView.postMessage(JSON.stringify(rpcRequest));
  }
})
// some bundler script that takes output of above and stringifies it
```

now, providers doesn't know about platform concerns